src/nvim/channel.c

```diff
@@ -626,8 +626,8 @@ static void channel_process_exit_cb(Process *proc, int status, void *data)
 {
   Channel *chan = data;
   if (chan->term) {
-    char msg[sizeof("\r\n[Process exited ]") + NUMBUFLEN];
-    snprintf(msg, sizeof msg, "\r\n[Process exited %d]", proc->status);
+    char msg[sizeof("\r\nexit status . Press ENTER to continue.") + NUMBUFLEN];
+    snprintf(msg, sizeof msg, "\r\nExit status %d. Press ENTER to continue.", proc->status);
```

Comment:
Maybe this is a good time to get rid of this message entirely and expose it as `v:shell_error` and/or  `v:event.exitcode` (for `TermClose`).

Reply:
Yes this was mainly a quick hack to get the "resonable" behavior specifically for `:!` working.  I would still like `:terminal` to display a message if the process exited non-zero or by a deadly signal, but maybe this will be possible with a (possibly default) autocmd rather than something hardcoded.